sdk/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsClient.cs

```diff
@@ -2072,6 +2072,451 @@ private Response<RecognizeLinkedEntitiesResultCollection> RecognizeLinkedEntitie
+        {
+            Argument.AssertNotNullOrEmpty(documents, nameof(documents));
+            options ??= new AnalyzeHealthOptions();
+            MultiLanguageBatchInput documentInputs = ConvertToMultiLanguageInputs(documents, language);
+
+            return StartAnalyzeHealth(documentInputs, options, cancellationToken);
+        }
+
+        /// <summary>
+        /// Recognizes health entities async.
+        /// </summary>
+        /// <param name="documents"></param>
+        /// <param name="options"></param>
+        /// <param name="cancellationToken">A <see cref="CancellationToken"/> controlling the request lifetime.</param>
+        /// <returns>A <see cref="AnalyzeHealthOperation"/> to wait on this long-running operation.  Its <see cref="AnalyzeHealthOperation.Value"/> upon successful
+        /// completion will contain layout elements extracted from the form.</returns>
+        public virtual AnalyzeHealthOperation StartAnalyzeHealth(IEnumerable<TextDocumentInput> documents, AnalyzeHealthOptions options, CancellationToken cancellationToken = default)
+        {
+            Argument.AssertNotNull(documents, nameof(documents));
+
+            options ??= new AnalyzeHealthOptions();
+
+            MultiLanguageBatchInput documentInputs = ConvertToMultiLanguageInputs(documents);
+
+            using DiagnosticScope scope = _clientDiagnostics.CreateScope($"{nameof(TextAnalyticsClient)}.{nameof(StartAnalyzeHealth)}");
+            scope.Start();
+
+            try
+            {
+                ResponseWithHeaders<TextAnalyticsHealthHeaders> response = _serviceRestClient.Health(documentInputs, options.ModelVersion, _stringCodeUnit, cancellationToken);
+                string location = response.Headers.OperationLocation;
+
+                return new AnalyzeHealthOperation(_serviceRestClient, _clientDiagnostics, location, documentInputs);
+            }
+            catch (Exception e)
+            {
+                scope.Failed(e);
+                throw;
+            }
+        }
+
+        /// <summary>
+        /// Recognizes layout elements from one or more passed-in forms.
+        /// </summary>
+        /// <param name="documents"></param>
+        /// <param name="options"></param>
+        /// <param name="cancellationToken">A <see cref="CancellationToken"/> controlling the request lifetime.</param>
+        /// <returns>A <see cref="AnalyzeHealthOperation"/> to wait on this long-running operation.  Its <see cref="AnalyzeHealthOperation.Value"/> upon successful
+        /// completion will contain layout elements extracted from the form.</returns>
+        public virtual async Task<AnalyzeHealthOperation> StartAnalyzeHealthAsync(IEnumerable<TextDocumentInput> documents, AnalyzeHealthOptions options, CancellationToken cancellationToken = default)
+        {
+            Argument.AssertNotNull(documents, nameof(documents));
+
+            options ??= new AnalyzeHealthOptions();
```

Comment:
to avoid repeating code, wouldn't it be better to just call `StartAnalyzeHealth(MultiLanguageBatchInput batchInput ...)` and keep this method short?

Reply:
Same comment for the sync method